text/src/autogluon/text/automm/predictor.py

```diff
@@ -433,67 +427,135 @@ def fit(
         self._eval_metric_name = eval_metric_name  # In case eval_metric isn't provided in __init__().
         self._validation_metric_name = validation_metric_name
         self._save_path = save_path
-        self._config = config
         self._output_shape = output_shape
         self._column_types = column_types
         self._df_preprocessor = df_preprocessor
-        self._data_processors = data_processors
-        self._model = model
-        self.mixup_fn = mixup_fn
-
-        # save artifacts for the current running, except for model checkpoint, which will be saved in _fit()
-        self.save(save_path)
-
-        if time_limit == timedelta(seconds=0):
-            self._top_k_average(
-                model=model,
-                save_path=save_path,
-                minmax_mode=minmax_mode,
-                is_distill=False,
-                config=config,
-                val_df=tuning_data,
-                validation_metric_name=validation_metric_name,
-            )
-            return self
+        
 
-        # need to assign the above attributes before setting up distillation
-        if teacher_predictor is not None:
-            teacher_model, critics, baseline_funcs, soft_label_loss_func, \
-                teacher_df_preprocessor, teacher_data_processors = \
-                self._setup_distillation(
-                    teacher_predictor=teacher_predictor,
-                )
-        else:
-            teacher_model, critics, baseline_funcs, soft_label_loss_func,\
-                teacher_df_preprocessor, teacher_data_processors = None, None, None, None, None, None
-
-        self._fit(
+        _fit_args = dict(
             train_df=train_data,
             val_df=tuning_data,
-            df_preprocessor=df_preprocessor,
-            data_processors=data_processors,
-            model=model,
-            config=config,
-            loss_func=loss_func,
             validation_metric=validation_metric,
```

Comment:
Do we need `validation_metric` and `custom_metric_func ` to initialize HPO? It seems that only `validation_metric_name` is used.